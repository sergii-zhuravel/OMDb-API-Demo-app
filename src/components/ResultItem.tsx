import React from "react";
import styled from 'styled-components';

const StyledImage = styled.img`
  width: 50px;
`;

const CenteredCell = styled.td`
  text-align: center;
`;

export type ResultItemProps = {
    Poster: string;
    Title: string;
    Year: string;
    imdbID: string;
}

export default function ResultItem({ Title, Year, Poster }: ResultItemProps) {
    return <tr>
        <td>{Title}</td>
        <CenteredCell>{Year}</CenteredCell>
        <CenteredCell><StyledImage src={Poster} /></CenteredCell>
    </tr>
}