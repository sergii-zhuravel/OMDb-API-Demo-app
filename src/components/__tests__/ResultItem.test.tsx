import * as React from 'react';
import { shallow } from 'enzyme';
import ResultItem from '../ResultItem';

test('ResultItem component test', () => {
    const component = shallow(<ResultItem Title="Title" Year="2021" Poster="test/url" imdbID="1" />);

    // Snapshot 
    expect(component).toMatchSnapshot();
});