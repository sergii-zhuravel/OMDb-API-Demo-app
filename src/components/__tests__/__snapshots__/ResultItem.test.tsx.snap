// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultItem component test 1`] = `
<tr>
  <td>
    Title
  </td>
  <styled.td>
    2021
  </styled.td>
  <styled.td>
    <styled.img
      src="test/url"
    />
  </styled.td>
</tr>
`;
